lib/Analysis/AxisInfo.cpp

```diff
@@ -1220,6 +1220,19 @@ void AxisInfo::initPessimisticStateFromFunc(int argNumber, T funcOp,
 
 unsigned ModuleAxisInfoAnalysis::getContiguity(Value value) {
   auto tensorTy = dyn_cast<RankedTensorType>(value.getType());
+  if (!tensorTy)
+    return 1;
+  auto elemTy = tensorTy.getElementType();
+  // Get the pointee type if we have a tensor of ptrs to compute contiguity for
+  if (auto ptrTy = dyn_cast<PointerType>(elemTy)) {
+    elemTy = ptrTy.getPointeeType();
+  }
+  return getContiguity(value, elemTy.getIntOrFloatBitWidth());
+}
+
+unsigned ModuleAxisInfoAnalysis::getContiguity(Value value,
+                                               unsigned elementBitWidth) {
+  auto tensorTy = dyn_cast<RankedTensorType>(value.getType());
```

Comment:
I think we want to `cast` instead of `dyn_cast` here? By taking a separate `elementBitWidth` in this overload, it implicitly assumes `value` is a tensor value no?